apps/web/src/components/common/TokenAmountInput/index.tsx

```diff
@@ -1,56 +1,93 @@
+import NumberField from '@/components/common/NumberField'
+import { AutocompleteItem } from '@/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer'
 import { safeFormatUnits } from '@/utils/formatters'
+import { validateDecimalLength, validateLimitedAmount } from '@/utils/validation'
 import { Button, Divider, FormControl, InputLabel, MenuItem, TextField } from '@mui/material'
 import { type SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
-import css from './styles.module.css'
-import NumberField from '@/components/common/NumberField'
-import { validateDecimalLength, validateLimitedAmount } from '@/utils/validation'
-import { AutocompleteItem } from '@/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer'
-import { useFormContext } from 'react-hook-form'
 import classNames from 'classnames'
 import { useCallback } from 'react'
+import { get, useFormContext } from 'react-hook-form'
+import css from './styles.module.css'
+import { MultiTokenTransferFields, type MultiTokenTransferParams } from '@/components/tx-flow/flows/TokenTransfer'
 
 export enum TokenAmountFields {
   tokenAddress = 'tokenAddress',
   amount = 'amount',
 }
 
+export const InsufficientFundsValidationError = 'Insufficient funds'
+
+const getFieldName = (field: TokenAmountFields, groupName?: string) => (groupName ? `${groupName}.${field}` : field)
+
 const TokenAmountInput = ({
   balances,
   selectedToken,
   maxAmount,
   validate,
+  groupName,
+  onChangeAmount = () => {},
```

Comment:
Nit: we don't need to assign a default here. We can call this conditionally: `onChangeAmount?.()`.